### PR TITLE
Remove error-prone {short,long}_term_capital_losses variables

### LINKED ISSUE #2622: Remove the {short,long}_term_capital_losses variables
There are four person-level income variables related to capital gains:
* `short_term_capital_gains`
* `short_term_capital_losses`
* `long_term_capital_gains`
* `long_term_capital_losses`

The documentation of these variables does not make it clear (at least to me) how to use these variables.   I'm assuming that the four variables are all supposed to be non-negative in value and then the code computes net gains (which can be positive or negative).   But that assumption is not correct.  Proceeding under that assumption leads to incorrect AGI results (as shown in the tests below).  Ignoring the capital losses variables and entering a loss as a negative gain does produce correct AGI results.

These tests show that the `{short,long}_capital_losses` variables are redundant and that there existence is dangerous in the sense that using them will produce incorrect results.

Here are the two simple tests, the first of which passes and the second of which fails giving an `adjusted_gross_income` value of 100_000, which is clearly incorrect.

```
- name: capital_losses integration test 1
  absolute_error_margin: 0.01
  period: 2018
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 40
        employment_income: 100_000
        long_term_capital_gains: -5_000
    tax_units:
      tax_unit:
        members: [person1]
    spm_units:
      spm_unit:
        members: [person1]
    households:
      household:
        members: [person1]
        state_code: TX  # skip state income tax calculations
  output:
    adjusted_gross_income: 97_000

- name: capital_losses integration test 2
  absolute_error_margin: 0.01
  period: 2018
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 40
        employment_income: 100_000
        long_term_capital_losses: 5_000
    tax_units:
      tax_unit:
        members: [person1]
    spm_units:
      spm_unit:
        members: [person1]
    households:
      household:
        members: [person1]
        state_code: TX  # skip state income tax calculations
  output:
    adjusted_gross_income: 97_000
```

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #2622.

### Summary
📝🐛♻️

This pull request refactors and fixes the variables and parameters related to capital losses and gains in the US tax system. It removes redundant and unused variables, renames and updates existing variables to match the IRS logic and terminology, and adds a new variable for capital losses at the person level. It also updates the metadata, documentation, and tests accordingly.

> _We are the masters of the capital gains_
> _We crush the errors and the losses with our brains_
> _We use the `limited_capital_loss` to deduct_
> _We make the formulas simple and robust_

### Walkthrough
*  Rename and refactor the variable `maximum_capital_loss` to `limited_capital_loss`, which represents the capital loss that is limited by the parameter `gov.irs.ald.loss.capital.max` ( [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-49c66a919a6fe5b3fa48711f1794ef1bc91bbcf7f83ead3de398e12ca7952168L4-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-49c66a919a6fe5b3fa48711f1794ef1bc91bbcf7f83ead3de398e12ca7952168L14-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-20de81a830a399c8c722658f869785799c2da434c4058a92acc9c6ee6be5773aL18-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-239c7951a1e79847ad904b0a2e835e600a32e28d2797d864f3e36431924fd2f8L42-R42), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-364ca8beaf5168f302c99a2c3a4dd93a40d7f911dc14fe5d94a72fb8f790169cL17-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-fdace9a68f8d093826b60c76290ef71867f04b779d6d34920f2fa17f80e5698fR1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-e2d7342138e16adb697fe35828ccb24156ef8b166f9b6b8b409f8ecdcd639c70))
*  Update the formula of `net_capital_gain`, which sums up the qualified dividend income and the long-term and short-term capital gains for each tax unit member, using the `add` and `max_` functions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-64875cb8918107f5b1aadfe3f5d2e8b3497083309d4794798d31757deaaf4590L17-R29), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-188e0d973f45bdb629ee6c198a7dad6a0fe67b5f21949fba8e094d1ddb13942d), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-2d22694b928755da5d95d62b2e13cfbcf1e70b3dd93fff0f208b44dc1261e58e), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-dad7ba5c59dfac899fa5be7576f2fc2ee58b868a8957c246ea614aa2d6d14017), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-45072d144c210a4c589b3eecf203a324e9493193d97e5ca430b7a35376059abf))
*  Add metadata to the parameter `gov.irs.ald.loss.capital.max`, which defines the maximum capital loss deductible above-the-line for different filing statuses ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-7e59a5b00e86fbfd98e7644f009037ace9c4975a3c0838b33c7b921f6451b5f9R2-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-7e59a5b00e86fbfd98e7644f009037ace9c4975a3c0838b33c7b921f6451b5f9L12-L18))
*  Update the label and documentation of `capital_gains`, `long_term_capital_gains`, and `short_term_capital_gains`, which calculate the net capital gains and losses for each person, clarifying that losses are expressed as negative gains ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-ae72cc6aa7b6ce1cb90a422f1e23f277e1137e7216370c739ed3ad6e45016d9cL7-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-978926be7f0b3c301f2ab2bee9a8cdd6cf805e91bee0ae86ce790dcacd2829ccL7-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-7830ac8c5dc6d1ae16a5fa9b909f12aee2ee1f146fa6cc5fed637e53d62253d8L7-R11))
*  Add a new variable `capital_losses`, which calculates the net capital losses for each person as the opposite of `capital_gains`, and is useful for calculating the limited capital loss deduction at the tax unit level ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-fdace9a68f8d093826b60c76290ef71867f04b779d6d34920f2fa17f80e5698fR1-R13))
*  Update the value of `is_tax_unit_head` for `person1` in the test file `irs_gross_income.yaml`, which tests the calculation of the IRS gross income, to match the expected boolean type of the variable ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-8aa69f7763cdf70e79fa03d89b65165f8f77e9e0d2483a3689b11fd7b1e43ee2L17-R17))
*  Simplify the calculation of the self-employment loss in the formula of `loss_ald`, which calculates the above-the-line deduction for losses, by using the `person` alias and the `max_` function ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-20de81a830a399c8c722658f869785799c2da434c4058a92acc9c6ee6be5773aL18-R22))
*  Simplify the calculation of the nonbusiness losses in the formula of `mo_ptc_gross_income`, which calculates the gross income for the Missouri Property Tax Credit, by using the `add` function ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-239c7951a1e79847ad904b0a2e835e600a32e28d2797d864f3e36431924fd2f8L42-R42))
*  Reformat the documentation of `net_capital_gain`, by breaking the long string into multiple lines and adding parentheses ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-64875cb8918107f5b1aadfe3f5d2e8b3497083309d4794798d31757deaaf4590L9-R13))
*  Simplify the access to the parameter `gov.irs.ald.loss.capital.max` in the formula of `limited_capital_loss`, by using the alias `p` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-49c66a919a6fe5b3fa48711f1794ef1bc91bbcf7f83ead3de398e12ca7952168L14-R17))
*  Add a changelog entry to the `changelog_entry.yaml` file, indicating that the pull request fixes an error-prone use of variables related to capital losses and bumps the patch version of the package ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Delete unused variables related to capital gains and losses, such as `capital_loss`, `long_term_capital_losses`, and `short_term_capital_losses` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-9736be6feee73ccd70c340989eaaca43bc54acc2cbf3cc7023e4e04c222a1d78), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-6279ce86a8db079642ec0d89e2ea670e412f3f857d0673e0156b725d2c23b62b), [link](https://github.com/PolicyEngine/policyengine-us/pull/2654/files?diff=unified&w=0#diff-6484bf33fbca18e3e7d7299f772aa1860ef16c0ef36256e363dcbcc5af122c4d))

### CODE REVIEW

**Comment by martinholmer** [comment]
The reduction in code coverage in PR #2654 is caused by the fact that the `excess` function (in the `policyengine_us/tools/general.py` file) is no longer used anywhere in the code.

**Comment by MaxGhenis** [comment]
@nikhilwoodruff could you review this given you implemented it initially? Also could we remove the `excess` function?

File: policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/limited_capital_loss.py [hunk 1]
```diff
@@ -1,19 +1,17 @@
 from policyengine_us.model_api import *
 
 
-class maximum_capital_loss(Variable):
+class limited_capital_loss(Variable):
```
Comment (nikhilwoodruff) [inline policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/limited_capital_loss.py:4]: OK I understand now- thanks for correcting this variable name.

**Review by nikhilwoodruff — APPROVED** [review]
This looks good, thanks @martinholmer

**Pull request merged**